ui-core/src/components/inputs/ComboBox/ComboBox.tsx

```diff
@@ -80,15 +80,26 @@ const ComboBox = <T,>({
 
     // Call the parent's onChange handler with the synthetic event
     onChange?.(syntheticEvent);
+  };
+
+  const clearAndFocusInput = () => {
+    clearInputValue();
     focusInput();
   };
 
+  /* eslint-disable react-hooks/exhaustive-deps */
+  useEffect(() => {
+    if (!value) {
+      clearInputValue();
+    }
+  }, [value]);
```

Comment:
Can you modify the existing hook line 120 instead of adding a new one ?
Also, why do you call clearInputValue ? setValue(value) is not enough ?:thinking: